### PR TITLE
Use spawner with `--params-file` argument instead of cli verbs

### LINKED ISSUE #395: Demos and tests do not work
As a consequence of https://github.com/ros-controls/ros2_control/pull/1694 the parameters are not known to the spawners without explicitely giving `--params-file` option.
This make the tests fail, something in the shutdown of gz hangs and there are still joint_states published in later tests. 
https://github.com/ros-controls/ros2_control_ci/issues/126
https://github.com/ros-controls/ros2_control_ci/issues/127

maybe I have time to fix this in the next days.

### PR DESCRIPTION
**Pull request by christophfroehlich**
This repo is now in line with the ros2_control_demo repo, especially https://github.com/ros-controls/ros2_control_demos/pull/502

I tested all examples and fixed some other stuff (wrong topic for ackermann steering controller etc). I'll create a manual backport for iron.

Fixes #395

### CODE REVIEW

**Comment by christophfroehlich** [comment]
@mergifyio backport jazzy

File: gz_ros2_control_demos/config/cart_controller_velocity.yaml [hunk 1]
```diff
@@ -2,21 +2,16 @@ controller_manager:
...
-
     joint_state_broadcaster:
       type: joint_state_broadcaster/JointStateBroadcaster
 
-    imu_sensor_broadcaster:
-      type: imu_sensor_broadcaster/IMUSensorBroadcaster
-
 velocity_controller:
   ros__parameters:
     joints:
```
Comment (saikishor) [inline gz_ros2_control_demos/config/cart_controller_velocity.yaml:10]:
```suggestion
   ros__parameters:
      type: velocity_controllers/JointGroupVelocityController
     joints:
```

**Review by saikishor — COMMENTED** [review]
Thanks for the nice work. The changes till now looks good to me

File: gz_ros2_control_demos/config/cart_controller_velocity.yaml (hunk 1 above)
Comment (christophfroehlich) [reply gz_ros2_control_demos/config/cart_controller_velocity.yaml]: oops

**Review by ahcorde — APPROVED** [review]
Thank you! Tested!

File: gz_ros2_control_demos/config/cart_controller_velocity.yaml (hunk 1 above)
Comment (saikishor) [reply gz_ros2_control_demos/config/cart_controller_velocity.yaml]: Thanks

**Review by saikishor — APPROVED** [review]
LGTM

**Pull request merged**